### ISSUE TITLE
Allow Users to Hide Specific Content (e.g. Gay, Straight, Trans, BDSM) when using Stash-DB

### ISSUE THREAD

**Issue by randemgame** [stashapp/stash-box#643]
**Is your feature request related to a problem? Please describe.**
Using Stash-DB right now is an awkward experience if you do not want to be exposed to certain genres of pornography.

**Describe the solution you'd like**
I think a site-wide filtering system is the most traditional, tested and functional way to implement this.

**Describe alternatives you've considered**
There is discussion on Discord about this topic as of 05/05/23 in #stashdb-general

**Additional context**

I am building on an existing GitHub issue (from a now deleted user) https://github.com/stashapp/stash-box/issues/364 that has had limited interest. I am arguing my case here based on a Discord discussion that due to the lack of interest of that GitHub issue, <sub><sup>which doesn't have the most descriptive title or description</sup></sub>, that "this doesn't seem to bother a lot of people, so probably won't be a priority."

I am an average user who would like to contribute more to stash-db, adding and appending entries and fixing details in my personal Stash via the Stash-DB integration. I enjoy voting on issues and enjoy the Stash + StashDB user experience as an amazing resource to compliment their collection. 
I do not really enjoy looking at hardcore screenshots of penetrative sex that don't align with my sexuality, which as of 06/05 is quite dominant in the 'Scenes' tab and 'Edits' tab.
I think myself and other potential users may be turned off from contributing because they would rather not see such content that didn't align with their sexuality.
(from another Discord user: "It’s always ideal to reduce friction and lower the “barrier to entry” to get more people contributing.")

I thought that the Scene tags could be used to filter content - I noticed some scenes marked as Straight and some marked as Gay.
Scruffy says that not every scene is going to be tagged so that's not a good filter system.
But these sexual orientation tags already exist on scenes in the system. Over time, will more scenes be tagged with them, or less?
As long as Stash is making a database of every porn file in existence, I think it's feasible that much of it could be tagged by orientation over time, as inherited by Studio or added/modified by user.
I expect this is dev logic I don't understand. Could scenes not tagged as such have a placeholder tag of 'Unknown Orientation'. Maybe users could be invited to browse through scenes with an Unknown Orientation and correct any missing information? I am just spitballing ideas now! I would do some time sorting through that to apologies for some overly dramatic language in my first post.
So with DogmaDragon saying below that there is an issue with untagged scenes and scenes with gender-neutral tags - I think it could be an imperfect system that would be improved over time with user input. Gender is difficult but I'm sure a great majority of content could fall into a main sort of descriptor of Gay, Trans or Straight.

I would love to scroll the 'Edits' tab with some filter attached so I could look at what people are uploading and vote on it or add comments etc while double-checking data and stuff, while avoiding the content I was not interested in. This could also be an opportunity to tag scenes missing a sexual orientation?

Sorry for the wall of text. **TL:DR**; _Please add the possibility to filter out scenes and stars that the user is not interested in - specifically Gay/Trans/Straight/BDSM, but preferably inclusive of all tags - when browsing Scenes, Homepage or Edits

EDIT: Apologies for the dramatic and inappropriate language in the first post, some of which DogmaDragon has replied to below. I have edited the query to be much shorter and specific, and I added some new thoughts. But I realise I am speaking from a position of ignorance about these this site development and have annoyed people in the process inadvertently. Sorry!

**Comment by DogmaDragon** [comment]
First of all, this issue doesn't offer a viable solution. There is no way to know if the scene is gay/straight/trans based on existing attributes. Tags could be used, but there are plenty of scenes that are untagged, but even tagged scenes often use gender-neutral tags. **Below is my opinion.**

> Thinking about this some more, I would like a solution that would go beyond implementing filters as previously requested here (https://github.com/stashapp/stash-box/issues/364) to somewhat limited interest. I am arguing my case here for this system based on a comment on Discord that due to the lack of interest in that GitHub issue, which doesn't have the most descriptive title or description, that "this doesn't seem to bother a lot of people, so probably won't be a priority."

It's open-source software, anyone can add a feature and submit a pull request. What _"this doesn't seem to bother a lot of people, so probably won't be a priority."_ meant was that Stash Core Team are likely to prioritize issues that are [most requested](https://github.com/stashapp/stash-box/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc), have a bounty or are something that developers are interested themselves. The core aspect of open-source development is that like-minded people are working on new features they are interested in and then share them with others. 

> Imagine a user who would like to contribute more to stash-db, adding and appending entries, fixing details, voting on issues and in return enjoying the website as an amazing resource to compliment their collection.
Now imagine that user does not really enjoy looking at hardcore screenshots of penetrative sex that don't align with their sexuality. As they continue using the site they grow kind of frustrated to be constantly exposed to such content, typically when they click the 'Scenes' tab or when they scroll through the 'Edits' tab.
Does that user attempt to communicate their displeasure on Discord? Do they spend an inordinate amount of time crafting an eloquent GitHub issue describing their dissatisfaction?

To be completely honest I don't see an issue here. If someone is interested in contributing there are plenty of ways to do that without ever going to the Edits page. They can go to work on the performers that align with their sexuality, browse studios they like and work on those studios/scenes from those studios, search scenes by tags and only work on those scenes.

Adding new content has no impact on what StashDB shows or not show so I'm unsure how that is related. 

> Probably not. They just won't use it. Users will pick the path of least resistance.
(from another Discord user: "It’s always ideal to reduce friction and lower the “barrier to entry” to get more people contributing.")

There seems to be a general misunderstanding what StashDB is and what is meant to accomplish. It's a community database tailored for Stash software to query to get the metadata. Any registered user can use it without ever coming back to the site once they grab their API key. 

This is probably controversial, but I prefer 1 active contributor who takes time to read the guidelines and is diligent than 10 contributors who are looking to do tasks with the least resistance just to say they did. 

---

I'm ignoring the rant portion of the issue. 

---

> Why I have not just boosted the previous smol comment on filters is that I want this to be considered as something that has to be a site wide measure. Any implementation of filters is likely going to target the main Home/Scenes interface and I worry that the 'Edits' tab might be overlooked, which would not resolve the heart of this issue - users would still have to scroll through graphic content they are uncomfortable with in order to contribute to the DB. I would infinitely scroll the Edits tab if it did not show me gay content. Ideally, you could have the submissions that pop up there checked against the associated scene/star tags such as Gay, Straight or Trans before being presented (or not) to a user. Look, if a gay scene were to show up on my Edit feed that was not tagged as such, I'm not going to shit the bed - I would tag it as Gay and that would hopefully then filter it from my page (also, would be nice if that's how 'abstain' worked now). I am not objecting to seeing that content I try to avoid around 5% of the time, it comes with the territory of being a porn hound, but right now using Stash, the objectionable content feels more like 50%. (Also, there's the potential user behavior in down-voting the content that makes them comfortable rather than assessing it on merit)

You can scroll Edits page without seeing gay content by selecting a Type different than scene. 
![image](https://user-images.githubusercontent.com/103123951/236603756-b77e4111-9659-443e-b25f-cbb52a4ba2b0.png)

Any user who would engage in abusive behaviour would lose their EDIT role. 

---

I'm ignoring the rant portion of the issue. 

---

For me the proposal goes against the idea of a community database. Every user can choose how much to contribute and areas to contribute to. Nobody has to keep voting on scenes if they are disturbing for them to see. They can skip through them or move their focus to other area that doesn't contain that disturbing content.

**Comment by randemgame** [comment]
Thanks for the feedback. I updated the post for clarity and added new words. 
This is the new addition:

I thought that the Scene tags could be used to filter content - I noticed some scenes marked as Straight and some marked as Gay.
Scruffy says that not every scene is going to be tagged so that's not a good filter system.
But these sexual orientation tags already exist on scenes in the system. Over time, will more scenes be tagged with them, or less?
As long as Stash is making a database of every porn file in existence, I think it's feasible that much of it could be tagged by orientation over time, as inherited by Studio or added/modified by user.
I expect this is dev logic I don't understand. Could scenes not tagged as such have a placeholder tag of 'Unknown Orientation'. Maybe users could be invited to browse through scenes with an Unknown Orientation and correct any missing information? I am just spitballing ideas now! I would do some time sorting through that to apologies for some overly dramatic language in my first post.
So with DogmaDragon saying below that there is an issue with untagged scenes and scenes with gender-neutral tags - I think it could be an imperfect system that would be improved over time with user input. Gender is difficult but I'm sure a great majority of content could fall into a main sort of descriptor of Gay, Trans or Straight.

I would love to scroll the 'Edits' tab with some filter attached so I could look at what people are uploading and vote on it or add comments etc while double-checking data and stuff, while avoiding the content I was not interested in. This could also be an opportunity to tag scenes missing a sexual orientation?

TL:DR; _Please add the possibility to filter out scenes and stars that the user is not interested in - specifically Gay/Trans/Straight/BDSM, but preferably inclusive of all tags - when browsing Scenes, Homepage or Edits

EDIT: Apologies for the dramatic and inappropriate language in the first post, some of which DogmaDragon has replied to below. I have edited the query to be much shorter and specific, and I added some new thoughts. But I realise I am speaking from a position of ignorance about these this site development and have annoyed people in the process inadvertently. Sorry!

**Comment by doguitar** [comment]
The vast majority of this request could be done by allowing the hiding of studios. It would require seeing the content once to hide it, however.

**Comment by VonTittyslappen** [comment]
> The vast majority of this request could be done by allowing the hiding of studios. It would require seeing the content once to hide it, however.

I don't believe that to be true, the workaround is to have a section in the user profile where they can type in the name of the studio they want to block and the search reads against a list of all studios and you can add them to the blocklist there. That way you don't have to see the content to block it

This is an important feature that should be added as a priority IMO. There are quite a number of people out there who want to block visibility of certain things and they should have the option to do so.

:)

**Comment by BonerFide** [comment]
Without reading all of the back and forth, and knowing the limitations, there are times I have to scroll through a whole page of scene edits which I'm utterly unqualified to review, and cannot filter out, in order to find edits I can meaningfully contribute to. Yes, it would be better if I could just not see those. 

No, that's not easy to do without putting a label on the scene, but as mentioned a default label/s for a studio ought make that easier. 

You look at enough porn you get good at visually ignoring things you don't really want to see, but it's still annoying and makes reviewing edits harder than it needs to be. It's not even about not seeing a particular gender or whatever, it's just if the type of studios they work for are that outside your zone of knowledge you're not going to have anything meaningful to add. 

Some things don't need a 100% solution, a 70% solution would help. I understand putting this in the 'too hard' basket. But I think dismissing it outright is not really being considerate of a fair chunk of the community. Also just what's shown on the homepage isn't filterable. 

There's 1000 features I'd rather see before this. But I understand it. The bigger the community gets the more it will be useful, especially as the edit queue grows and needs to attract more competent reviewers to manage the workload.

**Comment by Serechops** [comment]
Howdy all. I saw this brought up yet again in the Stash Discord, so I decided to write a Userscript for it. You'll need to utilize something like Tampermonkey in order to load it correctly.

Here's the link: https://github.com/Serechops/Serechops-Stash/tree/main/StashDB%20Userscript

I have a predefined list of Studios in there currently to get anyone started, but you can feel free to modify the list as your tastes see fit. 

This will apply a blur filter to the image card from defined Studios within that list.

Cheers!

**Comment by neru2132** [comment]
I think even outside of sexual preference a filter of user-defined ignored tags could be useful, say for hiding re-distributions, re-releases, compilations and so on.
Part of the issue with un-tagged or insufficiently tagged content could be alleviated by new/extended contribution guidelines and the upcoming notification system. The active community would probably just sort it out among themselves over time. It might even lead to better data quality as people seeing content that they would rather not see might be more inclined to write a comment encouraging the contributor to add the appropriate tag. In my estimation it would be an overall more straightforward solution to filter based on tags rather than a comprehensive and ever expanding list of all studios one is not interested in. What's more, in the case of very large studios, it's quite possible people are interested in part of the their content, but not all.

**Comment by echo6ix** [comment]
@neru2132 Yeah, that's not a bad idea if I understand correctly. But there's already an issue with rule creep, so this is the kind of thing that can be automated.

Submitting a new scene --> Onboarding tag section --> "Is this content `gay`, `straight`, `trans` etc". Click each tag that applies. Boom. Something like this gets a baseline small set of tags that orient the scenes fundamental genre.

It could also be accomplished with studio tags to an extent, though not all studios are so black and white with their content, but I suspect the vast majority are.

Having multiple approaches wouldn't hurt.
